src/main/resources/application.properties

```diff
@@ -2,4 +2,12 @@ quarkus.http.host=0.0.0.0
 quarkus.smallrye-openapi.info-description=Just an example service
 quarkus.smallrye-openapi.info-contact-email=techsupport@example.com
 quarkus.smallrye-openapi.info-contact-name=Example API Support
-quarkus.smallrye-openapi.info-contact-url=http://exampleurl.com/contact
\ No newline at end of file
+quarkus.smallrye-openapi.info-contact-url=http://exampleurl.com/contact
+# configure your datasource
+quarkus.datasource.db-kind = postgresql
+quarkus.datasource.username = hibernate
+quarkus.datasource.password = hibernate
+quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/hibernate
```

Comment:
- Deploying DB and running quarkus app

I deployed the postgresql instance as mentioned from you.

By running `quarkus dev` I receive following error

###

__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2022-12-04 17:23:26,742 WARN  [io.agr.pool] (agroal-11) Datasource '<default>': Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.

2022-12-04 17:23:26,746 WARN  [org.hib.eng.jdb.env.int.JdbcEnvironmentInitiator] (JPA Startup Thread: <default>) HHH000342: Could not obtain connection to query metadata: org.postgresql.util.PSQLException: Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:319)
        at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
        at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
        at org.postgresql.Driver.makeConnection(Driver.java:434)
        at org.postgresql.Driver.connect(Driver.java:291)
        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
        at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
        at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:253)
        at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.base/java.net.Socket.connect(Socket.java:609)
        at org.postgresql.core.PGStream.createSocket(PGStream.java:241)
        at org.postgresql.core.PGStream.<init>(PGStream.java:98)
        at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:109)
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
        ... 12 more


2022-12-04 17:23:26,909 WARN  [io.agr.pool] (agroal-11) Datasource '<default>': Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
2022-12-04 17:23:26,910 WARN  [org.hib.eng.jdb.spi.SqlExceptionHelper] (JPA Startup Thread: <default>) SQL Error: 0, SQLState: 08001
2022-12-04 17:23:26,911 ERROR [org.hib.eng.jdb.spi.SqlExceptionHelper] (JPA Startup Thread: <default>) Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
2022-12-04 17:23:26,933 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.net.ConnectException: Connection refused (Connection refused)
        at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
        at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:253)
        at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.base/java.net.Socket.connect(Socket.java:609)
        at org.postgresql.core.PGStream.createSocket(PGStream.java:241)
        at org.postgresql.core.PGStream.<init>(PGStream.java:98)
        at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:109)
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
        at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
        at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
        at org.postgresql.Driver.makeConnection(Driver.java:434)
        at org.postgresql.Driver.connect(Driver.java:291)
        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)


2022-12-04 17:23:26,936 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure

--
Tests paused
Press [space] to restart, [e] to edit command line args (currently ''), [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>

###

- Testing Connection

After testing the connection, like describe [here](https://stackoverflow.com/questions/26911508/postgres-testing-database-connection-in-bash) I receive following message:

###

root@4ffa65374055:/ball-bearings-with-quarkus# pg_isready -d hibernate -h 0.0.0.0 -p 5432 -U hibernate
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
0.0.0.0:5432 - no response
root@4ffa65374055:/ball-bearings-with-quarkus# pg_isready -d hibernate -h 127.0.0.1 -p 5432 -U hibernate
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
127.0.0.1:5432 - no response

###

- next Steps

How is your development set up ? Are you developing your app inside a docker container ? I think there is an Issue Connection the quarkus App from one container to the DB container.

Reply:
The Issue has to located inside the dev docker of the quarkus app.

By running `myUser@myMachine:~$ pg_isready -d hibernate -h 0.0.0.0 -p 5432 -U hibernate
0.0.0.0:5432 - accepting connections` from my host terminal. The connection works.

Reply:
I installed locale `apt-get -y install locales`

and set the locale

`locale-gen en_US.UTF-8 `
`update-locale LANG=en_US.UTF-8`

But the error still occurs. Now with a shorter response:

`root@4ffa65374055:/ball-bearings-with-quarkus# pg_isready -d hibernate -h 0.0.0.0 -p 5432 -U hibernate`

throws

`0.0.0.0:5432 - no response`

Reply:
I found the solution:

Connection from the dev container to the DB container works via the actual HostIP like

`pg_isready -d hibernate -h 192.168.178.27 -p 5432 -U hibernate`

throws

`192.168.178.27:5432 - accepting connections`

The result is, that we need to adapt the settings of the application.properties of the quarkus app and/or the linking of the container via docker network https://docs.docker.com/network/network-tutorial-standalone/

Reply:
Could you take care of the solution? I will review your changes as usual.

Reply:
Thank you for your troubleshooting!

I will work on the solution.

Reply:
I adjusted the hostname in `application.properties` via the env variable `POSTGRESQL_SERVICE_HOST`.

Copy the content of `.env.example` in your own .env file. When you are developing on your local machine the env variable has to be unset, when you are developing in a docker container the varibale has to be set to `hibernate`. You can do that via comments (#). 

The quarkus dev container and the database have to be in the same network to resolve hostnames, therefore do the following steps:
- `docker network create --driver bridge quarkus-net`
- attach already created container to the new network: `docker network connect quarkus-net <container-name>` 
 **or** create new container in the network: `docker build --build-arg GH_TOKEN=<GH-TOKEN> --build-arg username=<git-username>--build-arg email=<git-email> -t quarkus_image .` (unchanged) + `docker run --network quarkus-net --name quarkus -it quarkus_image`
- attach database container to the network: `docker run --rm=true --name hibernate --network quarkus-net -e POSTGRES_USER=hibernate -e POSTGRES_PASSWORD=hibernate -e POSTGRES_DB=hibernate -p 5432:5432 postgres:14.1`